firmware/laseramp.ino

```diff
@@ -1,143 +1,251 @@
+/*
+Release 0.1
+We ignore N commands.
+
+Todo:
+Support for N\d+ line prefix.
+Set on run time the size of bed, the steps per mm
+Save configuration to eeprom
+Save to eeprom default values.
+Finish Raster mode.
+Support for negative values in X, Y, Z.
+Add Homing option for HARDWARE positioning and WORK positioning
+
+http://www.deskcnc.com/Homing.html
+Raster Mode
+G5 <[0-9A-F]>
+each char is a 8bit long code, each command may send up to 64 bytes.cada
+*/
+
+
 #include "commandqueue.h"
 #include "motion.h"
 #include "globals.h"
+#include "const.h"
 #include "pins.h"
 #include "command.h"
-#include "const.h"
+#include "fastio.h"
+
+#if   MOTHERBOARD == 1 //RAMPS14
+#include <TimerThree.h>
+#endif
+#if   MOTHERBOARD == 2 //GRBL
+#include <TimerOne.h>
+#endif
+
+
+
+//void initialize_pins();
+
+void ISRTIMER5COMPAvect() {
+  ms_since_step[0]++;
+  ms_since_step[1]++;
+#ifdef Z_DRIVER_ENABLED
+  ms_since_step[2]++;
+#endif
+
+ #ifdef EASYCNCCONTROLLER
+   debug_counter0++;
+  if ( debug_counter0 == 1000 ) {
```

Comment:
This is probably pretty bad to do in an interrupt handler. Can we do this in the control loop instead?